ansible/roles/cuda/tasks/install.yml

```diff
@@ -29,17 +29,22 @@
   when: "'No matching Modules to list' in _cuda_driver_module_enabled.stderr"
   changed_when: "'Nothing to do' not in _cuda_driver_module_enable.stdout"
 
-- name: Check if nvidia driver module is installed
-  ansible.builtin.command: dnf module list --installed nvidia-driver
+- name: Read module info for list of packages
+  ansible.builtin.shell:
+    cmd: >-
+      dnf module info nvidia-driver:{{ cuda_nvidia_driver_stream }} |
+      grep -F {{ cuda_nvidia_driver_version }}.el{{ ansible_distribution_major_version }}.{{ ansible_architecture }}
```

Comment:
I think some packages don't have el9 suffix (distro independent) so does that mean this isn't a complete list?

Reply:
oh yeah like `nvidia-imex-570-0:570.124.06-1.x86_64`

So `dnf module info nvidia-driver:570-open` only appears to show packages ending .el9.x86_64, `el9.noarch` and `.x86_64`. So maybe its enough to just suffix the version with a `.` when grepping - I was trying to avoid the case where you are after e.g. `570-0:570.124.06-1` and just grepping for that also gets you `570-0:570.124.06-10`.

What do you think?

Reply:
Oh good spot, I'd missed those :facepalm:

Reply:
So might need to filter for x86_64 and noarch on top of that you suggested? :thinking:

Reply:
I guess we can just `| unique` the list?

Reply:
Doh no grep just version+ . obvs returns the entire package name

Reply:
or the other option is to just install the i686 packages too (not sure how much bloat that adds)

Reply:
Hmm, even with this we get:
```
Depsolve Error occurred: 
 Problem 1: conflicting requests
  - nothing provides cuda-drivers-570 = 570.133.20 needed by cuda-drivers-fabricmanager-570-570.133.20-1.x86_64 from cuda-rhel9-x86_64
 Problem 2: package cuda-drivers-fabricmanager-570.133.20-1.x86_64 from cuda-rhel9-x86_64 requires cuda-drivers-fabricmanager-570 = 570.133.20, but none of the providers can be installed
  - conflicting requests
  - nothing provides cuda-drivers-570 = 570.133.20 needed by cuda-drivers-fabricmanager-570-570.133.20-1.x86_64 from cuda-rhel9-x86_64
```